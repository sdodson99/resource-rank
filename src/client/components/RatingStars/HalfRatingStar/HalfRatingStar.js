import React from 'react';
import PropTypes from 'prop-types';
import styles from './HalfRatingStar.module.css';
import { StarHalfRounded } from '@material-ui/icons';

const HalfRatingStar = ({ size, color }) => (
  <div className={styles.HalfRatingStar} data-testid="HalfRatingStar">
    <StarHalfRounded htmlColor={color} style={{ fontSize: size }} />
  </div>
);

HalfRatingStar.propTypes = {
  size: PropTypes.number,
  color: PropTypes.string,
};

HalfRatingStar.defaultProps = {
  size: 20,
  color: 'black',
};

export default HalfRatingStar;
