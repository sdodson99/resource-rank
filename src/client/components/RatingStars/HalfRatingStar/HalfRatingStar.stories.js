import React from 'react';
import HalfRatingStar from './HalfRatingStar';

export default {
  component: HalfRatingStar,
  title: 'Components/RatingStars/HalfRatingStar',
  args: {
    size: 50,
  },
};

const Template = (args) => <HalfRatingStar {...args} />;

export const Default = Template.bind({});
Default.args = {};

export const Color = Template.bind({});
Color.args = {
  color: 'orange',
};

export const Size = Template.bind({});
Size.args = {
  ...Color.args,
  size: 100,
};
