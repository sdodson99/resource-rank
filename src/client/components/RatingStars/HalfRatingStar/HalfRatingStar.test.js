import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import HalfRatingStar from './HalfRatingStar';

describe('<HalfRatingStar />', () => {
  test('it should mount', () => {
    render(<HalfRatingStar />);
    
    const halfRatingStar = screen.getByTestId('HalfRatingStar');

    expect(halfRatingStar).toBeInTheDocument();
  });
});