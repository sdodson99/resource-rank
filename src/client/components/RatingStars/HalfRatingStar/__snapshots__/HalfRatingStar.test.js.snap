// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HalfRatingStar /> should render correctly 1`] = `
<div
  className="HalfRatingStar"
  data-testid="HalfRatingStar"
>
  <Memo
    htmlColor="black"
    style={
      Object {
        "fontSize": 20,
      }
    }
  />
</div>
`;

exports[`<HalfRatingStar /> should render correctly with custom color 1`] = `
<div
  className="HalfRatingStar"
  data-testid="HalfRatingStar"
>
  <Memo
    htmlColor="yellow"
    style={
      Object {
        "fontSize": 20,
      }
    }
  />
</div>
`;

exports[`<HalfRatingStar /> should render correctly with custom size 1`] = `
<div
  className="HalfRatingStar"
  data-testid="HalfRatingStar"
>
  <Memo
    htmlColor="black"
    style={
      Object {
        "fontSize": 100,
      }
    }
  />
</div>
`;
